From 083aed594af931118d06fef1d99167e6b4afb34d Mon Sep 17 00:00:00 2001
From: Santhana Kumar <santhanakumar.a@ltts.com>
Date: Fri, 5 Apr 2019 19:24:57 +0530
Subject: [PATCH 40/47] Adapt HDMI interface.

- Move HDMI from IPU1 DI0 to DI1 so it's no longer shared with parallel RGB
  but can display independent content.
- HDMI EDID is not on I2C2, we use the dedicated DDC function instead. Also
  means we can support HDCP (High-bandwidth Digital Content Protection)
  now. Replace seperate I2C EDID node with pinmux.
- Boost data signals D0-D2 to improve rise time: On LEC-iMX6 module plus
  carrier combos, signal quality is trickier than on SabreSD SBCs. To pass
  compliance set sup_tx_lvl and tx_{sym,tra,trb}on to maximum.
- HDMI CEC (Consumer Electronics Control) uses a different pin.
- The fb device is equally enabled for Quad and Solo, move 'status' to
  common DTSI.

Signed-off-by: Santhana Kumar <santhanakumar.a@ltts.com>
---
 arch/arm/boot/dts/lec-imx6.dtsi | 10 +++-------
 1 file changed, 3 insertions(+), 7 deletions(-)

diff --git a/arch/arm/boot/dts/lec-imx6.dtsi b/arch/arm/boot/dts/lec-imx6.dtsi
index 082423e..e86eca1 100644
--- a/arch/arm/boot/dts/lec-imx6.dtsi
+++ b/arch/arm/boot/dts/lec-imx6.dtsi
@@ -496,9 +496,9 @@
 	status = "okay";
 };
 
-&hdmi_core {
+&hdmi_core {		/* HDMI (see fb above) */
 	ipu_id = <0>;
-	disp_id = <0>;
+	disp_id = <1>;
 	status = "okay";
 };
 
@@ -689,10 +689,6 @@
 		};
 	};
 
-	hdmi_edid: edid@50 {
-		compatible = "fsl,imx6-hdmi-i2c";
-		reg = <0x50>;
-	};
 #if CONFIG_I2S_AUDIO == 1
 	codec: tlv320aic23@1a {		/* Audio codec on LEC-Base R1 carrier */
 		compatible = "ti,tlv320aic23";
@@ -869,7 +865,7 @@
 
 		pinctrl_hdmi_cec: hdmicecgrp {
 			fsl,pins = <
-				MX6QDL_PAD_KEY_ROW2__HDMI_TX_CEC_LINE 0x108b0
+				MX6QDL_PAD_EIM_A25__HDMI_TX_CEC_LINE	0x108b0
 			>;
 		};
 
-- 
2.7.4

